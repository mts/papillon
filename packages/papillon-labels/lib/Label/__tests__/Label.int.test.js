import {
  // regular
  labelVariant,
  labelWhiteOnBlueVariant,
  labelWhiteOnDarkGrayVariant,
  labelWhiteOnGreenVariant,
  labelWhiteOnRedVariant,
  labelWhiteOnYellowVariant,
  labelWhiteOnPurpleVariant,
  // theme
  themeGrayVariant,
  themeDarkGrayVariant,
  themeOrangeVariant,
  themeOutlineVariant,
  themeOutlineGreenVariant,
  // state
  stateVariant,
  stateDefaultVariant,
  stateGreenVariant,
  statePurpleVariant,
  stateRedVariant,
  stateDefaultSmallVariant,
  stateGreenSmallVariant,
  statePurpleSmallVariant,
  stateRedSmallVariant,
  stateOpenVariant,
  stateClosedVariant,
  stateMergedVariant,
  // counter
  counterVariant,
  counterDefaultVariant,
  counterGrayVariant,
  counterGrayLightVariant,
} from './Label.int.render'

describe('<Label />', () => {
  describe('Snaphot', () => {
    describe('regular', () => {
      test('must match labelVariant', () => {
        expect(global.renderToJSON(labelVariant)).toMatchSnapshot()
      })

      test('must match labelWhiteOnBlueVariant', () => {
        expect(global.renderToJSON(labelWhiteOnBlueVariant)).toMatchSnapshot()
      })

      test('must match labelWhiteOnDarkGrayVariant', () => {
        expect(global.renderToJSON(labelWhiteOnDarkGrayVariant)).toMatchSnapshot()
      })

      test('must match labelWhiteOnGreenVariant', () => {
        expect(global.renderToJSON(labelWhiteOnGreenVariant)).toMatchSnapshot()
      })

      test('must match labelWhiteOnRedVariant', () => {
        expect(global.renderToJSON(labelWhiteOnRedVariant)).toMatchSnapshot()
      })

      test('must match labelWhiteOnYellowVariant', () => {
        expect(global.renderToJSON(labelWhiteOnYellowVariant)).toMatchSnapshot()
      })

      test('must match labelWhiteOnPurpleVariant', () => {
        expect(global.renderToJSON(labelWhiteOnPurpleVariant)).toMatchSnapshot()
      })
    })

    describe('theme', () => {
      test('must match themeGrayVariant', () => {
        expect(global.renderToJSON(themeGrayVariant)).toMatchSnapshot()
      })

      test('must match themeDarkGrayVariant', () => {
        expect(global.renderToJSON(themeDarkGrayVariant)).toMatchSnapshot()
      })

      test('must match themeOrangeVariant', () => {
        expect(global.renderToJSON(themeOrangeVariant)).toMatchSnapshot()
      })

      test('must match themeOutlineVariant', () => {
        expect(global.renderToJSON(themeOutlineVariant)).toMatchSnapshot()
      })

      test('must match themeOutlineGreenVariant', () => {
        expect(global.renderToJSON(themeOutlineGreenVariant)).toMatchSnapshot()
      })
    })

    describe('state', () => {
      test('must match stateVariant', () => {
        expect(global.renderToJSON(stateVariant)).toMatchSnapshot()
      })

      test('must match stateDefaultVariant', () => {
        expect(global.renderToJSON(stateDefaultVariant)).toMatchSnapshot()
      })

      test('must match stateGreenVariant', () => {
        expect(global.renderToJSON(stateGreenVariant)).toMatchSnapshot()
      })

      test('must match statePurpleVariant', () => {
        expect(global.renderToJSON(statePurpleVariant)).toMatchSnapshot()
      })

      test('must match stateRedVariant', () => {
        expect(global.renderToJSON(stateRedVariant)).toMatchSnapshot()
      })

      test('must match stateDefaultSmallVariant', () => {
        expect(global.renderToJSON(stateDefaultSmallVariant)).toMatchSnapshot()
      })

      test('must match stateGreenSmallVariant', () => {
        expect(global.renderToJSON(stateGreenSmallVariant)).toMatchSnapshot()
      })

      test('must match statePurpleSmallVariant', () => {
        expect(global.renderToJSON(statePurpleSmallVariant)).toMatchSnapshot()
      })

      test('must match stateRedSmallVariant', () => {
        expect(global.renderToJSON(stateRedSmallVariant)).toMatchSnapshot()
      })

      test('must match stateOpenVariant', () => {
        expect(global.renderToJSON(stateOpenVariant)).toMatchSnapshot()
      })

      test('must match stateClosedVariant', () => {
        expect(global.renderToJSON(stateClosedVariant)).toMatchSnapshot()
      })

      test('must match stateMergedVariant', () => {
        expect(global.renderToJSON(stateMergedVariant)).toMatchSnapshot()
      })
    })

    describe('counter', () => {
      test('must match counterVariant', () => {
        expect(global.renderToJSON(counterVariant)).toMatchSnapshot()
      })

      test('must match counterDefaultVariant', () => {
        expect(global.renderToJSON(counterDefaultVariant)).toMatchSnapshot()
      })

      test('must match counterGrayVariant', () => {
        expect(global.renderToJSON(counterGrayVariant)).toMatchSnapshot()
      })

      test('must match counterGrayLightVariant', () => {
        expect(global.renderToJSON(counterGrayLightVariant)).toMatchSnapshot()
      })
    })
  })
})
