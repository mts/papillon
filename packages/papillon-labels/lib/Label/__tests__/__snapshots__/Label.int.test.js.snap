// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> Snaphot counter must match counterDefaultVariant 1`] = `
<span
  className="Counter"
>
   
  33
</span>
`;

exports[`<Label /> Snaphot counter must match counterGrayLightVariant 1`] = `
<span
  className="Counter Counter--gray-light"
>
   
  77
</span>
`;

exports[`<Label /> Snaphot counter must match counterGrayVariant 1`] = `
<span
  className="Counter Counter--gray"
>
   
  55
</span>
`;

exports[`<Label /> Snaphot regular must match labelWhiteOnBlueVariant 1`] = `
<span
  className="Label text-white bg-blue"
>
   
  papillon regular label
</span>
`;

exports[`<Label /> Snaphot regular must match labelWhiteOnDarkGrayVariant 1`] = `
<span
  className="Label text-white bg-gray-dark"
>
   
  papillon regular label
</span>
`;

exports[`<Label /> Snaphot regular must match labelWhiteOnGreenVariant 1`] = `
<span
  className="Label text-white bg-green"
>
   
  papillon regular label
</span>
`;

exports[`<Label /> Snaphot regular must match labelWhiteOnPurpleVariant 1`] = `
<span
  className="Label text-white bg-purple"
>
   
  papillon regular label
</span>
`;

exports[`<Label /> Snaphot regular must match labelWhiteOnRedVariant 1`] = `
<span
  className="Label text-white bg-red"
>
   
  papillon regular label
</span>
`;

exports[`<Label /> Snaphot regular must match labelWhiteOnYellowVariant 1`] = `
<span
  className="Label text-white bg-yellow"
>
   
  papillon regular label
</span>
`;

exports[`<Label /> Snaphot state must match stateClosedVariant 1`] = `
<span
  className="State State--red"
>
  <svg
    aria-hidden="true"
    className="octicon"
    height={16}
    role="img"
    style={
      Object {
        "display": "inline-block",
        "fill": "currentColor",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 12 16"
    width={12}
  >
    <path
      d="M11 11.28V5c-.03-.78-.34-1.47-.94-2.06C9.46 2.35 8.78 2.03 8 2H7V0L4 3l3 3V4h1c.27.02.48.11.69.31.21.2.3.42.31.69v6.28A1.993 1.993 0 0 0 10 15a1.993 1.993 0 0 0 1-3.72zm-1 2.92c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zM4 3c0-1.11-.89-2-2-2a1.993 1.993 0 0 0-1 3.72v6.56A1.993 1.993 0 0 0 2 15a1.993 1.993 0 0 0 1-3.72V4.72c.59-.34 1-.98 1-1.72zm-.8 10c0 .66-.55 1.2-1.2 1.2-.65 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2zM2 4.2C1.34 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"
      fillRule="evenodd"
    />
  </svg>
   
  papillon state label
</span>
`;

exports[`<Label /> Snaphot state must match stateDefaultSmallVariant 1`] = `
<span
  className="State State--small"
>
   
  papillon state label
</span>
`;

exports[`<Label /> Snaphot state must match stateDefaultVariant 1`] = `
<span
  className="State"
>
   
  papillon state label
</span>
`;

exports[`<Label /> Snaphot state must match stateGreenSmallVariant 1`] = `
<span
  className="State State--green State--small"
>
   
  papillon state label
</span>
`;

exports[`<Label /> Snaphot state must match stateGreenVariant 1`] = `
<span
  className="State State--green"
>
   
  papillon state label
</span>
`;

exports[`<Label /> Snaphot state must match stateMergedVariant 1`] = `
<span
  className="State State--purple"
>
  <svg
    aria-hidden="true"
    className="octicon"
    height={16}
    role="img"
    style={
      Object {
        "display": "inline-block",
        "fill": "currentColor",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 12 16"
    width={12}
  >
    <path
      d="M10 7c-.73 0-1.38.41-1.73 1.02V8C7.22 7.98 6 7.64 5.14 6.98c-.75-.58-1.5-1.61-1.89-2.44A1.993 1.993 0 0 0 2 .99C.89.99 0 1.89 0 3a2 2 0 0 0 1 1.72v6.56c-.59.35-1 .99-1 1.72 0 1.11.89 2 2 2a1.993 1.993 0 0 0 1-3.72V7.67c.67.7 1.44 1.27 2.3 1.69.86.42 2.03.63 2.97.64v-.02c.36.61 1 1.02 1.73 1.02 1.11 0 2-.89 2-2 0-1.11-.89-2-2-2zm-6.8 6c0 .66-.55 1.2-1.2 1.2-.65 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2zM2 4.2C1.34 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zm8 6c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"
      fillRule="evenodd"
    />
  </svg>
   
  papillon state label
</span>
`;

exports[`<Label /> Snaphot state must match stateOpenVariant 1`] = `
<span
  className="State State--green"
>
  <svg
    aria-hidden="true"
    className="octicon"
    height={16}
    role="img"
    style={
      Object {
        "display": "inline-block",
        "fill": "currentColor",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 12 16"
    width={12}
  >
    <path
      d="M11 11.28V5c-.03-.78-.34-1.47-.94-2.06C9.46 2.35 8.78 2.03 8 2H7V0L4 3l3 3V4h1c.27.02.48.11.69.31.21.2.3.42.31.69v6.28A1.993 1.993 0 0 0 10 15a1.993 1.993 0 0 0 1-3.72zm-1 2.92c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zM4 3c0-1.11-.89-2-2-2a1.993 1.993 0 0 0-1 3.72v6.56A1.993 1.993 0 0 0 2 15a1.993 1.993 0 0 0 1-3.72V4.72c.59-.34 1-.98 1-1.72zm-.8 10c0 .66-.55 1.2-1.2 1.2-.65 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2zM2 4.2C1.34 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"
      fillRule="evenodd"
    />
  </svg>
   
  papillon state label
</span>
`;

exports[`<Label /> Snaphot state must match statePurpleSmallVariant 1`] = `
<span
  className="State State--small State--purple"
>
   
  papillon state label
</span>
`;

exports[`<Label /> Snaphot state must match statePurpleVariant 1`] = `
<span
  className="State State--purple"
>
   
  papillon state label
</span>
`;

exports[`<Label /> Snaphot state must match stateRedSmallVariant 1`] = `
<span
  className="State State--small State--red"
>
   
  papillon state label
</span>
`;

exports[`<Label /> Snaphot state must match stateRedVariant 1`] = `
<span
  className="State State--red"
>
   
  papillon state label
</span>
`;

exports[`<Label /> Snaphot theme must match themeDarkGrayVariant 1`] = `
<span
  className="Label Label--gray-darker"
>
   
  papillon theme label
</span>
`;

exports[`<Label /> Snaphot theme must match themeGrayVariant 1`] = `
<span
  className="Label Label--gray"
>
   
  papillon theme label
</span>
`;

exports[`<Label /> Snaphot theme must match themeOrangeVariant 1`] = `
<span
  className="Label Label--orange"
>
   
  papillon theme label
</span>
`;

exports[`<Label /> Snaphot theme must match themeOutlineGreenVariant 1`] = `
<span
  className="Label Label--outline Label--outline-green"
>
   
  papillon theme label
</span>
`;

exports[`<Label /> Snaphot theme must match themeOutlineVariant 1`] = `
<span
  className="Label Label--outline"
>
   
  papillon theme label
</span>
`;
